ksql-cli/src/test/java/io/confluent/ksql/util/CliUtilsTest.java

```diff
@@ -16,30 +16,60 @@
 
 package io.confluent.ksql.util;
 
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import io.confluent.ksql.rest.entity.PropertiesList;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
 import org.junit.Test;
 
-import static org.junit.Assert.*;
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.Matchers.hasKey;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.fail;
 
-import java.io.IOException;
-import java.net.ConnectException;
-import java.sql.SQLDataException;
+import java.util.Collections;
+import java.util.Map;
 
 /**
  * Unit tests for class {@link CliUtils}.
  *
  * @see CliUtils
  */
 public class CliUtilsTest {
-
   @Test
   public void testGetAvroSchemaThrowsKsqlException() {
-    CliUtils cliUtils = new CliUtils();
-
     try {
+      CliUtils cliUtils = new CliUtils();
       cliUtils.getAvroSchema("TZGUM?ploV");
       fail("Expecting exception: KsqlException");
     } catch (KsqlException e) {
-      assertEquals(CliUtils.class.getName(), e.getStackTrace()[0].getClassName());
+      assertThat(CliUtils.class.getName(), equalTo(e.getStackTrace()[0].getClassName()));
     }
   }
+
+  @Test
+  public void shouldUpdateOverwrittenPropertiesCorrectly() {
+    final String key = KsqlConfig.KSQL_STREAMS_PREFIX + ConsumerConfig.AUTO_OFFSET_RESET_CONFIG;
+    final PropertiesList serverPropertiesList = new PropertiesList(
+        "list properties;",
+        ImmutableMap.of(key, "earliest"),
+        ImmutableList.of(key)
+    );
+    final Map<String, Object> properties = CliUtils.propertiesListWithOverrides(serverPropertiesList);
+
+    assertThat(properties, hasKey(key + " (LOCAL OVERRIDE)"));
+    assertThat(properties.get(key + " (LOCAL OVERRIDE)"), equalTo("earliest"));
+  }
+
+  @Test
+  public void shouldNotChangeUnwrittenProperty() {
```

Comment:
Isn't this just checking that unwritten properties are not considered overrides?

Reply:
Yes - it checks that we don't print (LOCAL OVERRIDE) for configs that haven't been overwritten.